src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2186,7 +2186,7 @@ private static IEnumerable<Diagnostic> FilterDiagnosticsByLocation(IEnumerable<D
             // Before returning diagnostics, we filter warnings
             // to honor the compiler options (/nowarn, /warnaserror and /warn) and the pragmas.
             var result = DiagnosticBag.GetInstance();
-            FilterAndAppendAndFreeDiagnostics(result, ref builder);
+            FilterAndAppendAndFreeDiagnostics(result, ref builder, includeDiagnosticsWithSourceSuppression);
```

Comment:
Just noticed this -- .NET guidelines would have you put input parameters like `includeDiagnosticsWithSourceSuppression` before `ref` or `out` parameters.